dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/CompanyRolesControllerTest.kt

```diff
@@ -384,17 +384,51 @@ class CompanyRolesControllerTest {
         }
         jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(TechnicalUser.Reader)
         val exceptionWhenTryingToGetCompanyRoles = assertThrows<ClientException> {
-            getCompanyRoleAssignments(CompanyRole.Member, companyIdAlpha)
+            getCompanyRoleAssignments(CompanyRole.Member, companyId = companyIdAlpha)
         }
         assertErrorCodeInCommunityManagerClientException(exceptionWhenTryingToGetCompanyRoles, 403)
         val exceptionWhenTryingToCheckCompanyRoles = assertThrows<ClientException> {
             hasUserCompanyRole(CompanyRole.DataUploader, companyIdAlpha, dataUploaderUserId)
         }
         assertErrorCodeInCommunityManagerClientException(exceptionWhenTryingToCheckCompanyRoles, 403)
     }
+
+    @Test
+    fun `assure that existing role assignments are deleted when a user gets assigned a new company role`() {
+        var currentAssignments: List<CompanyRoleAssignment>
+        val companyId = uploadCompanyAndReturnCompanyId()
+
+        jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(TechnicalUser.Admin)
+        assignCompanyRole(CompanyRole.DataUploader, companyId, dataUploaderUserId)
+
+        currentAssignments = getCompanyRoleAssignments(companyId = companyId, userId = dataUploaderUserId)
+        assertEquals(1, currentAssignments.size)
+        assertEquals(
+            CompanyRoleAssignment(
+                companyRole = CompanyRole.DataUploader,
+                companyId = companyId.toString(),
+                userId = dataUploaderUserId.toString(),
+            ),
+            currentAssignments.first(),
+        )
+
+        assignCompanyRole(CompanyRole.Member, companyId, dataUploaderUserId)
+
+        currentAssignments = getCompanyRoleAssignments(companyId = companyId, userId = dataUploaderUserId)
+        assertEquals(1, currentAssignments.size)
+        assertEquals(
+            currentAssignments.first(),
```

Comment:
One could think about switching the inputs as not to get confused what is actually expected when  the assertion fails. Could also be worthwile to write a small function inside of the method
```
fun assertRoleAssignments(role, currentAssignments)
```